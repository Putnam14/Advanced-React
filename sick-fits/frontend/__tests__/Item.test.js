import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import ItemComponent from '../components/Item'

const fakeItem = {
  id: 'ABC123',
  title: 'A cool item',
  price: 5000,
  description: 'This item is really cool!',
  image: 'dog.jpeg',
  largeImage: 'largedog.jpeg',
}

describe('<Item />', () => {
  it('renders the component', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
  it('renders the image', () => {
    // Shallow render the Item component
    const wrapper = shallow(<ItemComponent item={fakeItem} />)
    const Image = wrapper.find('img')
    expect(Image.props().src).toBe(fakeItem.image)
    expect(Image.props().alt).toBe(fakeItem.title)
  })

  it('renders the title and price tag', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />)
    const PriceTag = wrapper.find('PriceTag')
    expect(PriceTag.children().text()).toBe('$50')
    expect(wrapper.find('Title a').text()).toBe(fakeItem.title)
  })

  it('renders out the buttons', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />)
    const buttonList = wrapper.find('.buttonList')
    expect(buttonList.children()).toHaveLength(3)
    // Three ways to see if a tag exists:
    expect(buttonList.find('Link')).toHaveLength(1)
    expect(buttonList.find('Link').exists()).toBe(true)
    expect(buttonList.find('Link')).toBeTruthy()

    expect(buttonList.find('AddToCart').exists()).toBe(true)
    expect(buttonList.find('DeleteItem').exists()).toBe(true)
  })
})
