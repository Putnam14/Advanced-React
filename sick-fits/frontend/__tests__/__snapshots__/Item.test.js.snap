// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders the component 1`] = `
<ItemStyles__Item>
  <img
    alt="A cool item"
    src="dog.jpeg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        A cool item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $50
  </PriceTag>
  <p>
    This item is really cool!
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "/update",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        Edit ✏️
      </a>
    </Link>
    <AddToCart
      id="ABC123"
    />
    <DeleteItem
      id="ABC123"
    >
      Delete this item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
